classlib/src/main/java/org/teavm/classlib/java/util/TArrays.java

```diff
@@ -885,27 +885,17 @@ private static void merge(double[] a, double[] b, int from, int split, int to) {
     }
 
     public static void sort(Object[] a) {
-        sort(a, new NaturalOrder());
+        sort(a, TComparator.NaturalOrder.instance());
     }
 
     public static void sort(Object[] a, int fromIndex, int toIndex) {
-        sort(a, fromIndex, toIndex, new NaturalOrder());
-    }
-
-    private static class NaturalOrder implements TComparator<Object> {
-        @SuppressWarnings({ "rawtypes", "unchecked" })
-        @Override public int compare(Object o1, Object o2) {
-            if (o1 != null) {
-                return ((TComparable) o1).compareTo(o2);
-            } else if (o2 != null) {
-                return ((TComparable) o2).compareTo(o1);
-            } else {
-                return 0;
-            }
-        }
+        sort(a, fromIndex, toIndex, TComparator.NaturalOrder.instance());
```

Comment:
It isn't compiling in this case. Comparator.naturalOrder() has signature <T extends Comparable<? super T>> Comparator<T> naturalOrder() but in Arrays you require just Comparator<? super T> c without T extends Comparable. This is the reason why I moved it to separate instance class.

Reply:
Ah, I see. However, I still believe that you can use lambda instead of introducing NaturalOrder class. Under the hood TeaVM would create something like this NaturalOrder, but Java code would be more brief.